tdrs-backend/tdpservice/users/permissions.py

```diff
@@ -13,7 +13,11 @@ def is_own_stt(request, view):
         else None
     )
 
-    return is_data_prepper and user_stt and (requested_stt in [None, user_stt])
+    return (
+        is_data_prepper and
+        user_stt and
+        (requested_stt in [None, str(user_stt)])
```

Comment:
The casting to `str` here prevents some type errors that can occur depending on which endpoint is using this permission.